src/middlewared/middlewared/plugins/ssh.py

```diff
@@ -190,18 +192,40 @@ def save_keys(self):
             "ssh_host_ed25519_key",
             "ssh_host_ed25519_key.pub",
             "ssh_host_ed25519_key-cert.pub",
-        ]:
-            if osc.IS_FREEBSD:
-                path = os.path.join("/usr/local/etc/ssh", i)
-            else:
-                path = os.path.join("/etc/ssh", i)
+        ]
+    ]
+
+    @private
+    def cleanup_keys(self):
+        config = self.middleware.call_sync("datastore.config", "services.ssh")
+        for path, column in self.keys:
+            if not config[column] and os.path.exists(path):
+                self.middleware.logger.warning("Removing irrelevant SSH host key %r", path)
+                os.unlink(path)
```

Comment:
Should we catch an error here and log it if `unlink` fails for whatever reason?

Reply:
I can't imagine a reason it fails. We are writing/deleting files on rootfs all over middleware without any exception handling.